zrml/neo-swaps/src/lib.rs

```diff
@@ -648,53 +686,72 @@ mod pallet {
             min_amounts_out: Vec<BalanceOf<T>>,
         ) -> DispatchResult {
             ensure!(pool_shares_amount != Zero::zero(), Error::<T>::ZeroAmount);
-            let _ = T::MarketCommons::market(&market_id)?;
+            let market = T::MarketCommons::market(&market_id)?;
             Pools::<T>::try_mutate_exists(market_id, |maybe_pool| {
                 let pool =
                     maybe_pool.as_mut().ok_or::<DispatchError>(Error::<T>::PoolNotFound.into())?;
-                ensure!(
-                    pool.liquidity_shares_manager.fees == Zero::zero(),
-                    Error::<T>::OutstandingFees
-                );
-                let ratio = bdiv(
-                    pool_shares_amount.saturated_into(),
-                    pool.liquidity_shares_manager.total_shares()?.saturated_into(),
-                )?;
+                let ratio = {
+                    let mut ratio = pool_shares_amount
+                        .bdiv_floor(pool.liquidity_shares_manager.total_shares()?)?;
+                    if market.status == MarketStatus::Active {
+                        let multiplier = ZeitgeistBase::<BalanceOf<T>>::get()?
+                            .checked_sub_res(&EXIT_FEE.saturated_into())?;
+                        ratio = ratio.bmul_floor(multiplier)?;
+                    }
+                    ratio
+                };
```

Comment:
> I don't get this. For any other state than Active, the ratio does not include an EXIT_FEE?

Yes, exit fees are only charged while the market is active.

Reply:
> You use bdiv_floor here. AFAIK this operation needs to be combined with a ZeitgeistBase multiplication as you did with multiplier for the Active state. Suppose we don't have the Active state, where is the multiplication with ZeitgeistBase?

Not quite sure what you mean, but multiplying with `ZeitgeistBase`, which is one, won't accomplish anything.

Reply:
> Not quite sure what you mean, but multiplying with ZeitgeistBase, which is one, won't accomplish anything.

I mean this is the implementation of `bdiv`:

```rust
    fn bdiv_floor(&self, other: Self) -> Result<Self, DispatchError> {
        self.checked_mul_res(&ZeitgeistBase::get()?)?.checked_div_res(&other)
    }
```

I thought, you multiply the `ZeitgeistBase` here for precision purposes. So, I would have expected to see a division by `ZeitgeistBase` again. Means simplified this:

bdiv: `(a * BASE) / b` for better precision, because suppose `a = 5` and `b = 3`, you wanna avoid the result `1`, instead you wanna have `1.6666666667`. Assume BASE is `1000`, then you would get `5 * 1000 / 3 = 1666.666667`.

I thought you would have to divide this by `BASE` again, because the it's now a fixed-point representation with BASE. I say this, because I used bdiv and then bmul in the parimutuel implementation [here](https://github.com/zeitgeistpm/zeitgeist/blob/fb53170b99c57f5147ea7462247885ab6e36940e/zrml/parimutuel/src/lib.rs#L409-L416). So I assumed, you would always have a combination to first multiply the `ZeitgeistBase` for a higher precision and then later divide the value by the `ZeitgeistBase` to cancel it out again.

Reply:
So where did you expect the `ZeitgeistBase` factor? In the pallet or in the definition of `bmul_*`? I'm still not clear what the problem is.

Reply:
I am expecting to have a division by ZeitgeistBase after the `bdiv_floor` operation. I thought we would add precision, do the bmul or bdiv operation and then remove precision again. 

But I am open to learn that I am wrong.